lib/segment/src/index/hnsw_index/graph_links.rs

```diff
@@ -403,19 +403,25 @@ impl GraphLinksRam {
         let mut reindex: Vec<PointOffsetType> = Vec::new();
 
         let link_slice = get_links_slice(data, &header);
-        links.try_reserve_exact(link_slice.len())?;
+        let additional_capacity = link_slice.len().saturating_sub(links.capacity());
```

Comment:
I'm not sure that changes in this file are necessary. `try_reverse_exact` is called here for empty vector, so there is no matter, does it require capacity or additional size.